module/zfs/metaslab.c

```diff
@@ -1846,8 +1849,15 @@ metaslab_init(metaslab_group_t *mg, uint64_t id, uint64_t object, uint64_t txg,
 	 */
 	ms->ms_allocatable = range_tree_create_impl(&rt_avl_ops,
 	    &ms->ms_allocatable_by_size, metaslab_rangesize_compare, 0);
-	metaslab_group_add(mg, ms);
 
+	/*
+	 * The ms_trim tree is a subset of ms_allocatable which is kept
```

Comment:
Good comment. Let's move this block comment to the declaration of `ms_trim` member in `metaslab_impl.h`.

Reply:
I'm sure I'll understand this after finishing the code review, but... what if the metaslab is not loaded, in which case ms_allocatable is the empty set?

Reply:
The free ranges are added both to `ms_allocatable` and `ms_trim` in `metaslab_sync_done()` while the metaslab is loaded.  When the metaslab gets unloaded `ms_allocatable` gets vacated but the `ms_trim` tree is left populated for the autotrim thread to eventually consume.  Let me update the comment to make it clear its a subset of `ms_allocatable` if the metaslab were to be loaded.